src/pybind/mgr/cephadm/upgrade.py

```diff
@@ -494,6 +494,21 @@ def upgrade_stop(self) -> str:
         self.mgr.event.set()
         return 'Stopped upgrade to %s' % target_image
 
+    def update_service(self, service_type: str, service_image: str, image: str) -> List[str]:
+        exec(f"self.mgr.container_image_{service_image} = '{image}'")
```

Comment:
I feel like `setattr` passed the `self.mgr` object, `f"container_image_{service_image}"` setting and the `image` value would be a more natural way to do this than `exec` here as long as it works.

Reply:
As discussed using self.mgr.set_module_option to update image value.

```
ceph: root@ceph-node-0 /]# ceph orch ps --daemon-type=prometheus
NAME                    HOST         PORTS   STATUS        REFRESHED  AGE  MEM USE  MEM LIM  VERSION  IMAGE ID      CONTAINER ID  
prometheus.ceph-node-0  ceph-node-0  *:9095  running (4m)     3m ago   7m    30.2M        -  2.55.1   899b369f202e  c0280f7047c2  
prometheus.ceph-node-1  ceph-node-1  *:9095  running (4m)     3m ago   8m    28.9M        -  2.55.1   899b369f202e  21f80c3d7ef1  
prometheus.ceph-node-2  ceph-node-2  *:9095  running (3m)     3m ago   7m    29.4M        -  2.55.1   899b369f202e  69afb357cad7  
[ceph: root@ceph-node-0 /]# ceph config get mgr
WHO     MASK  LEVEL     OPTION                                     VALUE                                                                                                RO
mgr           advanced  container_image                            quay.io/rh-ee-shbhosal/ceph@sha256:11aa4f6a073f6aa5db0cf02a0a7b2536326b81e6f8c14f00b95719937a00410f  * 
mgr           advanced  log_to_file                                true                                                                                                 * 
mgr           advanced  mgr/cephadm/container_image_node_exporter  quay.io/prometheus/node-exporter:v1.8.0                                                                
mgr           advanced  mgr/cephadm/container_image_prometheus     quay.io/prometheus/prometheus:v2.55.1                                                                * 
mgr           advanced  mgr/cephadm/container_init                 True                                                                                                 * 
mgr           advanced  mgr/cephadm/log_to_cluster_level           debug                                                                                                * 
global        advanced  mgr/cephadm/migration_current              7                                                                                                    * 
mgr           advanced  mgr/dashboard/PROMETHEUS_API_HOST          http://ceph-node-0.local:9095                                                                          
global        advanced  mgr/dashboard/ssl_server_port              8443                                                                                                   
mgr           advanced  mgr/orchestrator/orchestrator              cephadm                                                                                              * 
global        basic     mon_cluster_log_to_file                    true                                                                                                   
global        basic     public_network                             192.168.100.0/24                                                                                     * 
[ceph: root@ceph-node-0 /]# ceph orch redeploy prometheus
Scheduled to redeploy prometheus.ceph-node-0 on host 'ceph-node-0'
Scheduled to redeploy prometheus.ceph-node-1 on host 'ceph-node-1'
Scheduled to redeploy prometheus.ceph-node-2 on host 'ceph-node-2'
[ceph: root@ceph-node-0 /]# ceph orch ps --daemon-type=prometheus
NAME                    HOST         PORTS   STATUS        REFRESHED  AGE  MEM USE  MEM LIM  VERSION  IMAGE ID      CONTAINER ID  
prometheus.ceph-node-0  ceph-node-0  *:9095  running (5m)     3m ago   8m    30.2M        -  2.55.1   899b369f202e  c0280f7047c2  
prometheus.ceph-node-1  ceph-node-1  *:9095  running (4m)     3m ago   8m    28.9M        -  2.55.1   899b369f202e  21f80c3d7ef1  
prometheus.ceph-node-2  ceph-node-2  *:9095  running (4m)     3m ago   8m    29.4M        -  2.55.1   899b369f202e  69afb357cad7  
[ceph: root@ceph-node-0 /]#
```